File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/java_solutions/1854-maximum-population-year.java
class Solution {
    public int maximumPopulation(int[][] logs) {
        int[] years = new int[2051];
        for (int[] log : logs) {
            years[log[0]]++;
            years[log[1]]--;
        }

        int maxPopulation = 0;
        int maxYear = 1950;
        int currentPopulation = 0;

        for (int year = 1950; year <= 2050; year++) {
            currentPopulation += years[year];
            if (currentPopulation > maxPopulation) {
                maxPopulation = currentPopulation;
                maxYear = year;
            }
        }

        return maxYear;
    }
}

File: MATTHEW/leetcode/AIcoding/4.GPT-4o/rust_solutions/1854-maximum-population-year.rs
impl Solution {
    pub fn maximum_population(logs: Vec<Vec<i32>>) -> i32 {
        let mut year_population = vec![0; 2051];

        for log in logs {
            year_population[log[0] as usize] += 1;
            year_population[log[1] as usize] -= 1;
        }

        let mut max_population = 0;
        let mut max_year = 0;
        let mut current_population = 0;

        for year in 1950..=2050 {
            current_population += year_population[year as usize];
            if current_population > max_population {
                max_population = current_population;
                max_year = year;
            }
        }

        max_year
    }
}